lib/r10k/git/shellgit.rb

```diff
@@ -53,26 +74,22 @@ def clean(opts = {})
         end
       end
 
-      def fetch(remote, opts = {})
-        cmd = ["fetch", remote, '+refs/*:refs/*']
-        result = git(cmd, opts)
-
-        if result.success?
-          return true
-        else
-          raise R10K::Git::GitError.new(result.stderr)
+      def clone(remote, local, opts={})
+        if opts[:private_key] || opts[:username]
+          logger.warn("Shellgit provider does not support custom SSH transport options, using default username and/or private key.")
         end
-      end
 
-      def clone(remote, local, opts={})
-        cmd = ["clone", "--mirror"]
+        cmd_opts = {
+          raise_on_fail: false,
+        }
+
+        cmd = ["clone", nil, "--mirror", remote, local]
+
         if opts[:bare]
-          cmd << "--bare"
+          cmd[1] = "--bare"
         end
```

Comment:
What's the reason for this being different than the others in the command construction?